gcloud-java-dns/src/main/java/com/google/gcloud/dns/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1266 @@
+        } else if (!"identity".equals(encoding)) {
+          throw new IOException(
+              "The request has the following unsupported HTTP content encoding: " + encoding);
+        }
+      }
+      return new String(ByteStreams.toByteArray(input), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      throw new IOException("Exception encountered when decoding request content.", e);
+    }
+  }
+
+  /**
+   * Generates a JSON response.
+   *
+   * @param context managedZones | projects | rrsets | changes
+   */
+  @VisibleForTesting
+  static Response toListResponse(List<String> serializedObjects, String context, String pageToken,
+      boolean includePageToken) {
+    StringBuilder responseBody = new StringBuilder();
+    responseBody.append("{\"").append(context).append("\": [");
+    Joiner.on(",").appendTo(responseBody, serializedObjects);
+    responseBody.append(']');
+    // add page token only if exists and is asked for
+    if (pageToken != null && includePageToken) {
+      responseBody.append(",\"nextPageToken\": \"").append(pageToken).append('"');
+    }
+    responseBody.append('}');
+    return new Response(HTTP_OK, responseBody.toString());
+  }
+
+  /**
+   * Prepares DNS records that are created by default for each zone.
+   */
+  private static ImmutableSortedMap<String, ResourceRecordSet> defaultRecords(ManagedZone zone) {
+    ResourceRecordSet soa = new ResourceRecordSet();
+    soa.setTtl(21600);
+    soa.setName(zone.getDnsName());
+    soa.setRrdatas(ImmutableList.of(
+        // taken from the service
+        "ns-cloud-c1.googledomains.com. cloud-dns-hostmaster.google.com. 0 21600 3600 1209600 312"
+    ));
+    soa.setType("SOA");
+    ResourceRecordSet ns = new ResourceRecordSet();
+    ns.setTtl(21600);
+    ns.setName(zone.getDnsName());
+    ns.setRrdatas(zone.getNameServers());
+    ns.setType("NS");
+    String nsId = getUniqueId(ImmutableSet.<String>of());
+    String soaId = getUniqueId(ImmutableSet.of(nsId));
+    return ImmutableSortedMap.of(nsId, ns, soaId, soa);
+  }
+
+  /**
+   * Returns a list of four nameservers randomly chosen from the predefined set.
+   */
+  @VisibleForTesting
+  static List<String> randomNameservers() {
+    ArrayList<String> nameservers = Lists.newArrayList(
+        "dns1.googlecloud.com", "dns2.googlecloud.com", "dns3.googlecloud.com",
+        "dns4.googlecloud.com", "dns5.googlecloud.com", "dns6.googlecloud.com"
+    );
+    while (nameservers.size() != 4) {
+      int index = ID_GENERATOR.nextInt(nameservers.size());
+      nameservers.remove(index);
+    }
+    return nameservers;
+  }
+
+  /**
+   * Returns a hex string id (used for a dns record) unique within the set of ids.
+   */
+  @VisibleForTesting
+  static String getUniqueId(Set<String> ids) {
+    String id;
+    do {
+      id = Long.toHexString(System.currentTimeMillis())
+          + Long.toHexString(Math.abs(ID_GENERATOR.nextLong()));
+      if (!ids.contains(id)) {
```

Comment:
Forgot to comment on this yesterday: can you remove this if and only return the ID at the end of the loop?